<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="UTF-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="generator" content="Asciidoctor 2.0.15.dev">
<title>Converting from 23andMe to VCF</title>
<link rel="stylesheet" href="./index.css">
</head>
<body class="article">
<div id="header">
</div>
<div id="content">
<div class="sidebarblock navig">
<div class="content">
<div class="ulist">
<div class="title">General</div>
<ul>
<li>
<p><a href="index.html">Main page</a></p>
</li>
<li>
<p><a href="../bcftools.html">Manual page</a></p>
</li>
<li>
<p><a href="install.html">Installation</a></p>
</li>
<li>
<p><a href="publications.html">Publications</a></p>
</li>
</ul>
</div>
<div class="ulist">
<div class="title">Calling</div>
<ul>
<li>
<p><a href="cnv-calling.html">CNV calling</a></p>
</li>
<li>
<p><a href="csq-calling.html">Consequence calling</a></p>
</li>
<li>
<p><a href="consensus-sequence.html">Consensus calling</a></p>
</li>
<li>
<p><a href="roh-calling.html">ROH calling</a></p>
</li>
<li>
<p><a href="variant-calling.html">Variant calling and filtering</a></p>
</li>
</ul>
</div>
<div class="ulist">
<div class="title">Tips and Tricks</div>
<ul>
<li>
<p><a href="convert.html">Converting formats</a></p>
</li>
<li>
<p><a href="annotate.html">Adding annotation</a></p>
</li>
<li>
<p><a href="query.html">Extracting information</a></p>
</li>
<li>
<p><a href="filtering.html">Filtering expressions</a></p>
</li>
<li>
<p><a href="scaling.html">Performance and Scaling</a></p>
</li>
<li>
<p><a href="plugins.html">Plugins</a></p>
</li>
<li>
<p><a href="FAQ.html">FAQ</a></p>
</li>
</ul>
</div>
</div>
</div>
<div id="main">
<div class="sect1">
<h2 id="_converting_from_23andme_to_vcf">Converting from 23andMe to VCF</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The raw 23andMe results can be downloaded as a tab-delimited file with four columns,
the marker ID, chromosome name, position and the genotype:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>rs6139074       20      63244   AA
rs1418258       20      63799   CC
rs6086616       20      68749   TT
rs6039403       20      69094   AG</pre>
</div>
</div>
<div class="paragraph">
<p>The dot &#8220;.&#8221; can be used for a marker ID in case it is not known. The conversion
command is then:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>bcftools convert --tsv2vcf input.tab.gz -f ref.fa -s SampleName -Ob -o sample.bcf</pre>
</div>
</div>
<div class="paragraph">
<p>It is important to check the output printed on the screen, which may look for example
like this:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>Rows total:     612647
Rows skipped:   4751
Missing GTs:    20525
Hom RR:     318339
Het RA:     165598
Hom AA:     103420
Het AA:     14</pre>
</div>
</div>
<div class="paragraph">
<p>Here the program converted more than 95% of the rows, in the example 20525
genotypes are missing (&#8220;--&#8221;) and 4751 sites were skipped because the tool
only considers SNPs and ignores deletions and insertions.</p>
</div>
<div class="paragraph">
<p>One should also check the number of non-reference heterozygous genotypes, which
was 14 in this example. It should be small like this, because large number of
heterozygous alts (<code>Het AA</code>) indicates that the input alleles are not on the
forward strand. We can check this explicitly using the <a href="plugin.fixref.html">fixref plugin</a>.</p>
</div>
<div class="sect2">
<h3 id="_merging_into_multi_sample_vcfs">Merging into multi-sample VCFs</h3>
<div class="paragraph">
<p>Single-sample VCFs created in the previous step can be merged into one multi-sample VCF using the
following commands. The input files must be indexed, then can be merged:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>bcftools index sampleA.bcf
bcftools index sampleB.bcf
bcftools merge -Ob -o output.bcf sampleA.bcf sampleB.bcf</pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_feedback">Feedback</h3>
<div class="paragraph">
<p>We welcome your feedback, please help us improve this page by 
either opening an <a href="https://github.com/samtools/bcftools/issues">issue on github</a> or <a href="https://github.com/samtools/bcftools/tree/gh-pages/howtos">editing it directly</a> and sending
a pull request.</p>
</div>
</div>
</div>
</div>
</div>
</div>
<div id="footer">
<div id="footer-text">
</div>
</div>
</body>
</html>